### PR TITLE
Interop with Gradle Kotlin

### PR DESCRIPTION
**Pull request by wyhasany**
As of version 1.3.4 pretty-jupiter plugin won't work correctly with
extension fields defined in `build.gradle.kts`. This PRs makes interop
possible. More over I've extended readme to be more friendly for kotlin
devs.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -16,13 +16,31 @@ JAVA_TOOL_OPTIONS="-Dfile.encoding=UTF8"
 ```
 
 Using the [plugins DSL](https://docs.gradle.org/current/userguide/plugins.html#sec:plugins_block):
+<details open>
+<summary>Groovy</summary>
+
 ```groovy
 plugins {
-  id 'com.github.joselion.pretty-jupiter' version 'x.x.x'
+    id 'com.github.joselion.pretty-jupiter' version 'x.x.x'
```
Comment (JoseLion) [inline README.md:24]:
```suggestion
  id 'com.github.joselion.pretty-jupiter' version 'x.x.x'
```

File: README.md [hunk 2]
```diff
@@ -16,13 +16,31 @@ JAVA_TOOL_OPTIONS="-Dfile.encoding=UTF8"
...
 ```
 
+</details>
+<details>
+<summary>Kotlin</summary>
+
+```kotlin
+plugins {
+    id("com.github.joselion.pretty-jupiter") version "x.x.x"
+}
```
Comment (JoseLion) [inline README.md:35]:
```suggestion
plugins {
  id("com.github.joselion.pretty-jupiter") version "x.x.x"
}
```

File: README.md [hunk 3]
```diff
@@ -38,6 +56,25 @@ buildscript {
...
+
+```kotlin
+buildscript {
+    repositories {
+        url = uri("https://plugins.gradle.org/m2/")
+    }
+    dependencies {
+        classpath("gradle.plugin.com.github.joselion.pretty-jupiter:pretty-jupiter:x.x.x")
+    }
+}
```
Comment (JoseLion) [inline README.md:71]:
```suggestion
buildscript {
  repositories {
    url = uri("https://plugins.gradle.org/m2/")
  }
  dependencies {
    classpath("gradle.plugin.com.github.joselion.pretty-jupiter:pretty-jupiter:x.x.x")
  }
}
```

File: README.md [hunk 4]
```diff
@@ -52,46 +89,118 @@ The plugin can be customized adding a `prettyJupiter` closure to your `build.gra
...
 ```groovy
 prettyJupiter {
-  duration {
-    enabled = true
-    threshold = 75
-  }
+    duration {
+        enabled = true
+        threshold = 75
+    }
```
Comment (JoseLion) [inline README.md:102]:
```suggestion
  duration {
    enabled = true
    threshold = 75
  }
```

File: README.md [hunk 5]
```diff
@@ -52,46 +89,118 @@ The plugin can be customized adding a `prettyJupiter` closure to your `build.gra
...
 
-  failure {
-    maxMessageLines = 15
-    maxTraceLines = 10
-  }
+    failure {
+        maxMessageLines = 15
+        maxTraceLines = 10
+    }
+}
```
Comment (JoseLion) [inline README.md:108]:
```suggestion
  failure {
    maxMessageLines = 15
    maxTraceLines = 10
  }
}
```

File: README.md [hunk 6]
```diff
@@ -52,46 +89,118 @@ The plugin can be customized adding a `prettyJupiter` closure to your `build.gra
...
+prettyJupiter {
+    duration {
+        enabled.set(true)
+        threshold.set(75)
+    }
+
+    failure {
+        maxMessageLines.set(15)
+        maxTraceLines.set(10)
+    }
```
Comment (JoseLion) [inline README.md:125]:
```suggestion
prettyJupiter {
  duration {
    enabled.set(true)
    threshold.set(75)
  }

  failure {
    maxMessageLines.set(15)
    maxTraceLines.set(10)
  }
```

File: README.md [hunk 7]
```diff
@@ -52,46 +89,118 @@ The plugin can be customized adding a `prettyJupiter` closure to your `build.gra
...
 
+
+
+<details open>
+<summary>Groovy</summary>
+
 ```groovy
 test {
-  useJUnitPlatform()
+    useJUnitPlatform()
```
Comment (JoseLion) [inline README.md:143]:
```suggestion
  useJUnitPlatform()
```

File: README.md [hunk 8]
```diff
@@ -52,46 +89,118 @@ The plugin can be customized adding a `prettyJupiter` closure to your `build.gra
...
 
-  testLogging {
-    exceptionFormat 'short'
-    events 'started', 'skipped', 'failed'
-  }
+    testLogging {
+        exceptionFormat 'short'
+        events 'started', 'skipped', 'failed'
+    }
```
Comment (JoseLion) [inline README.md:148]:
```suggestion
  testLogging {
    exceptionFormat 'short'
    events 'started', 'skipped', 'failed'
  }
```

File: README.md [hunk 9]
```diff
@@ -52,46 +89,118 @@ The plugin can be customized adding a `prettyJupiter` closure to your `build.gra
...
+import org.gradle.api.tasks.testing.logging.TestLogEvent.STARTED
+
+tasks {
+    test {
+        useJUnitPlatform()
+        testLogging {
+            exceptionFormat = SHORT
+            events = setOf(STARTED, SKIPPED, FAILED)
+        }
+    }
```
Comment (JoseLion) [inline README.md:169]:
```suggestion
tasks {
  test {
    useJUnitPlatform()
    testLogging {
      exceptionFormat = SHORT
      events = setOf(STARTED, SKIPPED, FAILED)
    }
  }
```

File: README.md [hunk 10]
```diff
@@ -52,46 +89,118 @@ The plugin can be customized adding a `prettyJupiter` closure to your `build.gra
...
+We only need Junit 5 configuration in `build.gradle(.kts)` file:
+
+
+<details open>
+<summary>Groovy</summary>
+
 ```groovy
 test {
-  useJUnitPlatform()
+    useJUnitPlatform()
```
Comment (JoseLion) [inline README.md:186]:
```suggestion
  useJUnitPlatform()
```

File: README.md [hunk 11]
```diff
@@ -52,46 +89,118 @@ The plugin can be customized adding a `prettyJupiter` closure to your `build.gra
...
+</details>
+<details>
+<summary>Kotlin</summary>
+
+```kotlin
+tasks {
+    test {
+        useJUnitPlatform()
+    }
+}
```
Comment (JoseLion) [inline README.md:199]:
```suggestion
tasks {
  test {
    useJUnitPlatform()
  }
}
```

File: src/main/groovy/com/github/joselion/prettyjupiter/PrettyJupiterExtension.groovy [hunk 12]
```diff
@@ -17,16 +21,12 @@ class PrettyJupiterExtension {
     this.failure = objects.newInstance(Failure)
   }
 
-  def duration(Closure closure) {
-    closure.resolveStrategy = Closure.DELEGATE_FIRST
-    closure.delegate = duration
-    closure()
+  void duration(Action<Duration> durationAction) {
+    durationAction.execute(duration)
```
Comment (JoseLion) [inline src/main/groovy/com/github/joselion/prettyjupiter/PrettyJupiterExtension.groovy:25]: 👍🏼

**Review by JoseLion — COMMENTED** [review]
@wyhasany thanks for your contribution. I really appreciate it! Let's use 2 spaces indentation to keep consistency on that project wide 🙂

**Comment by wyhasany** [comment]
@JoseLion I've applied all your remarks :)

**Review by JoseLion — APPROVED** [review]
Looks good! Thanks again, I'll release this as a patch version 🙂

**Pull request merged**